spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/oauth2/client/OAuth2LoginAutoConfiguration.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.security.oauth2.client;
+
+import java.net.URI;
+
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.autoconfigure.security.oauth2.OAuth2Properties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.annotation.web.configurers.oauth2.client.OAuth2LoginConfigurer;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for OAuth 2.0 / OpenID Connect 1.0 login support.
+ *
+ * @author Joe Grandja
+ * @since 2.0.0
+ * @see HttpSecurity#oauth2Login()
+ */
+@Configuration
+@ConditionalOnWebApplication(type = ConditionalOnWebApplication.Type.SERVLET)
+@ConditionalOnClass({EnableWebSecurity.class, ClientRegistration.class})
+@ConditionalOnMissingBean(WebSecurityConfigurerAdapter.class)
+@ConditionalOnBean(ClientRegistrationRepository.class)
+@AutoConfigureBefore(ClientRegistrationRepositoryAutoConfiguration.class)
```

Comment:
@jgrandja shouldn't this be `@AutoConfigureAfter(ClientRegistrationRepositoryAutoConfiguration.class)` because that's the one that provides the `ClientRegistrationRepository` bean? Also, I think for the `@ConditionalOnMissingBean(WebSecurityConfigurerAdapter.class)` to match, it would need to be auto-configured before `SecurityAutoConfiguration` so that the default `WebSecurityConfigurerAdapter` from `SpringBootWebSecurityConfiguration` does not kick in.

Reply:
Thanks for the heads up @mbhave.
Yeah I noticed that today as I was re-working the PR.

I changed it to:
```
@AutoConfigureBefore(SecurityAutoConfiguration.class)
@AutoConfigureAfter(ClientRegistrationRepositoryAutoConfiguration.class)
```
I'm going to push this sometime tomorrow after I finish up a couple more things.
Thanks!